lib/optimizely/helpers/validator.rb

```diff
@@ -132,6 +132,15 @@ def inputs_valid?(variables, logger = NoOpLogger.new, level = Logger::ERROR)
           variables.delete :user_id
         end
 
+        if variables.include? :variable_type
+          # Empty variable_type is a valid user ID.
```

Comment:
huh? This can you explain this?

Reply:
If `variables[:variable_type]` is anything other than a String or null value, it is invalid. Otherwise, it is valid, and so we remove `variable_type` from `variables`. That way, it does not go through the `value.is_a?(String) && !value.empty?` check on line 145.

Reply:
Got it. Thanks